app/Decsys/Controllers/WordlistsController.cs

```diff
@@ -173,12 +170,40 @@ public async Task<IActionResult> DeleteRule(string wordlistId, int ruleIndex)
             }
 
             await _service.DeleteRule(wordlistId, ruleIndex);
-            
+
             wordlist = _service.List(ownerId);
 
             return Ok(wordlist.Rules);
         }
 
+        [HttpPost("{wordlistId}")] 
+        [Authorize(Policy = nameof(AuthPolicies.IsSurveyAdmin))]
+        [SwaggerOperation("Added a custom word for the current user")]
+        [SwaggerResponse(200, "Custom Word Added.")]
+        [SwaggerResponse(401, "User is not authenticated")]
+        [SwaggerResponse(403, "User is not authorized to perform this operation")]
+        [SwaggerResponse(404, "Wordlist not found.")]
+        public async Task<IActionResult> AddCustomWord(string wordlistId, [FromBody] WordlistWord wordlistWord)
+        {
+            if (string.IsNullOrWhiteSpace(wordlistWord.Type) || string.IsNullOrWhiteSpace(wordlistWord.Word))
+            {
+                return BadRequest("Invalid type or word.");
+            }
+
+            wordlistWord.Type = wordlistWord.Type.ToLowerInvariant();
+
+            if (wordlistWord.Type != "noun" && wordlistWord.Type != "adjective")
```

Comment:
perhaps not in this PR but another one, you could make an `enum` for word types? Then you can reuse it everywhere and not compare magic string like here